### PR TITLE
Adds readme and publishes image to ECR

### PR DESCRIPTION
**Pull request by dianabarsan**
- increases mocha timeout to always allow e2e image to be built: https://github.com/medic/cht-upgrade-service/runs/7462514577?check_suite_focus=true 

https://github.com/medic/cht-upgrade-service/issues/6
https://github.com/medic/cht-upgrade-service/issues/7

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,102 @@
...
+
+#### Download and save CHT upgrade service docker-compose yml file.
+
+Todo steps??
+
+#### Export the environment variables:
+
+| Name                 | Required | Description                                                                                                        |
+|----------------------|----------|--------------------------------------------------------------------------------------------------------------------|
+| `DOCKER_CONFIG_PATH` | no       | Absolute path to your docker-config file to allow access to authenticated AWS ECR endpoints to pull private images |
```
Comment (garethbowen) [inline README.md:27]: Move this one down so the required variables are first. Also document what happens if you don't specify it, presumably you can only get public images?

File: README.md [hunk 2]
```diff
@@ -1 +1,102 @@
...
+
+#### Export the environment variables:
+
+| Name                 | Required | Description                                                                                                        |
+|----------------------|----------|--------------------------------------------------------------------------------------------------------------------|
+| `DOCKER_CONFIG_PATH` | no       | Absolute path to your docker-config file to allow access to authenticated AWS ECR endpoints to pull private images | 
+| `CHT_COMPOSE_PATH`   | yes      | Absolute path to the folder where the CHT docker-compose file is saved                                             | 
+| `COUCHDB_USER`       | yes      | CouchDb main admin account username                                                                                |
+| `COUCHDB_PASSWORD`   | yes      | CouchDb main admin account password                                                                                | 
+| `COUCHDB_SECRET`     | no       | CouchDb secret                                                                                                     |
```
Comment (garethbowen) [inline README.md:31]: What is this used for? What happens if you don't provide it?

File: README.md [hunk 3]
```diff
@@ -1 +1,102 @@
...
+| `COUCHDB_USER`       | yes      | CouchDb main admin account username                                                                                |
+| `COUCHDB_PASSWORD`   | yes      | CouchDb main admin account password                                                                                | 
+| `COUCHDB_SECRET`     | no       | CouchDb secret                                                                                                     |  
+| `COUCHDB_UUID`       | no | The uuid of the CouchDb Server                                                                                     |
+| `API_PORT`           | no | Defaults to 5988. The port on which CHT-Api is available on the network.                                           | 
+| `MARKET_URL_READ`    | no | Defaults to `https://staging.dev.medicmobile.org`. Points to the source of the CHT-Core staging server.            |                                                                |                                                                 |
+| `BUILDS_SERVER`      | no | Defaults to `_couch/builds`. Points to the source of the CHT-Core staging server database.                         |
```
Comment (garethbowen) [inline README.md:35]: Couldn't these be combined? Or is that a pain for testing where we need a different db?

File: test/publish/index.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+      exitCode ? reject(err || `Closed with exit code ${exitCode}`) : resolve(data);
+    });
+  });
+};
+
+(async () => {
+  try {
+    const tags = getImageTags();
+    const dockerfilePath = path.join(__dirname, '..', '..', 'Dockerfile');
+    await dockerCommand([`build -f ${dockerfilePath}`, ...tags.map(tag => `--tag ${tag}`), '.']);
```
Comment (garethbowen) [inline test/publish/index.js:69]: This line combined with line 37 makes this really complicated. I know what you want to do but it took me a while to convince myself it was going to work. For readability I think this can be improved if you remove line 37 and replace this line with...

```suggestion
    const dockerBuildArgs = [
      'build',
      '-f',
      dockerfilePath,
      ...tags.map(tag => `--tag ${tag}`),
      '.'
    ];
    await dockerCommand(dockerBuildArgs);
```

**Review by garethbowen — CHANGES_REQUESTED** [review]
Very nice. Some docs suggestions and one readability change but otherwise it looks great!

File: README.md (hunk 3 above)
Comment (dianabarsan) [reply README.md:35]: I guess they could be, but we'd have to change in CHT-Core first. 
I think we still need 2 "market_urls", even for external contributions. We had an e2e test that checked the upgrade process (with horti), and we need to download some version from the main staging server for that one.

Comment (garethbowen) [reply README.md]: Ok, leave it as is. Thanks for the explanation!

File: test/publish/index.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+};
+
+const getImageTags = () => {
+  const versions = getVersions();
+  const tags = versions.map(version => version.replace(/\/|_/g, '-'));
+  return tags.map(tag => `${getRepo()}/cht-upgrade-service:${tag}`);
+};
+
+const dockerCommand = (params) => {
+  const args = [ ...params.filter(param => param).map(param => param.split(' ')) ].flat();
```
Comment (garethbowen) [inline test/publish/index.js:37]: I think you can now drop this line, because we always pass an array of args now, right?

File: README.md [hunk 6]
```diff
@@ -1 +1,102 @@
...
+| `CHT_COMPOSE_PATH`   | yes      | Absolute path to the folder where the CHT docker-compose file is saved.                                                                                                                                                                      | 
+| `COUCHDB_USER`       | yes      | CouchDb main admin account username.                                                                                                                                                                                                         |
+| `COUCHDB_PASSWORD`   | yes      | CouchDb main admin account password.                                                                                                                                                                                                         | 
+| `COUCHDB_SECRET`     | yes      | Secret that is used by the CouchDB peers to communicate with each other and to generate authentication cookies. This field is mandatory if you are running CouchDb clustered mode to make the same authentication cookie valid on all nodes. |
```
Comment (garethbowen) [inline README.md:30]: This changed from "no" to "yes" in this edit, but the comment says it's only mandatory if clustered. Was that change intentional?

**Review by garethbowen — CHANGES_REQUESTED** [review]

File: README.md (hunk 6 above)
Comment (dianabarsan) [reply README.md:30]: I'm of two minds about this one. Right now, it gets a default value. What do you think?

Comment (garethbowen) [reply README.md:30]: It's fine. The comment explains it clearly enough. And it's better that people provide it when not needed, rather than ignore it when it is needed, so erring to "yes" is safer. Thanks!

**Review by garethbowen — APPROVED** [review]
Awesome! Thanks.
